blog/content/post/2020-02-18-ew2020-btg-demo.md

```diff
@@ -0,0 +1,212 @@
+may achieve it by locking SPI flash containing the firmware for example,
+however it does not protect from SPI desoldering attack. New SPI can contain
+maliciously modified firmware which compromises Root of Trust and may even
+install some backdoors, keyloggers etc. In order to protect ourselves (and our
+machines) from it we may use Boot Guard.
+
+---
+### How Boot Guard works?
+
+Intel Boot Guard is a feature that allows firmware verification before the
+first instructions gets executed. How was it achieved? This feature is
+supported by Intel ME, which instructs the processor to load an Authenticated
+Code Module (ACM) signed by Intel and validate the firmware (in short). During
+the verification process is utilizes the Key Manifest, Boot Policy Manifest
+(which are a part of Boot Guard feature) and the Firmware Interface Table
+(FIT). Processor before fetching the first firmware instructions searches for
+the FIT and applies microcode updates which are pointed by FIT. The microcode
+verifies the ACM, which verifies the Key Manifest, which verifies Boot Policy
+Manifest (so complicated right?). In the end the Boot Policy Manifest contains
+the hashes and signatures of the code regions in firmware that have to be
+protected and verified. After the firmware verification, ACM allows the
+processor to fetch first instructions to be executed. So despite the firmware
+is maliciously modified, it cannot execute if firmware verification fails.
+Because the manifests that hold one's signing keys are validated before
+firmware can execute. Furthermore the public key hash of the public key from
+Key Manifest can be burned into hardware fuses and locked permanently to ensure
+nobody expect the main key owner can approve a firmware signing key used to
+sign Boot Policy Manifest.
+
+The flow I have describe is one of the most restrictive profiles that Boot
+Guard offers - strict verification without remediation. There are less
+restrictive profile which allow the platform to keep running for 30 minutes if
+firmware verification fails in order to restore system and flash trusted
+firmware. But the production behavior of Boot Guard ensures that unverified
+firmware cannot execute on the processor and immediately shutdowns the
+platform.
+
+---
+### Where can I get Boot Guard for my platform?
+
+One has to either pay thousands of $ to some big firmware development companies
+to enable Boot Guard on your hardware or buy a high end machine which has Boot
+Guard enabled.
+
+That was true up till recently, where one of our customers -
+[Protectli](https://protectli.com/) - asked us to improve the security of his
+products called Vault with Boot Guard. Protectli sells compact network
+appliance devices running Intel processors. The most powerful of them have
+Intel 7th generation processors which are Boot Guard capable. As a matter of
+fact that 3mdeb did coreboot port for those platforms to coreboot, it was even
```

Comment:
@miczyg1 I believe logic is little bit messed in this sentence

Reply:
too much coreboot in coreboot, fixed